.editorconfig

```diff
@@ -230,3 +230,231 @@ dotnet_naming_style.I_style.capitalization = pascal_case
+dotnet_diagnostic.SA1017.severity = warning
+dotnet_diagnostic.SA1018.severity = warning
+dotnet_diagnostic.SA1019.severity = warning
+dotnet_diagnostic.SA1020.severity = warning
+dotnet_diagnostic.SA1021.severity = warning
+dotnet_diagnostic.SA1022.severity = warning
+dotnet_diagnostic.SA1023.severity = warning
+dotnet_diagnostic.SA1024.severity = warning
+dotnet_diagnostic.SA1025.severity = warning
+dotnet_diagnostic.SA1026.severity = warning
+dotnet_diagnostic.SA1027.severity = warning
+dotnet_diagnostic.SA1028.severity = warning
+
+# Ordering Rules (SA1200-) https://github.com/DotNetAnalyzers/StyleCopAnalyzers/blob/1.1.118/documentation/OrderingRules.md
+
+dotnet_diagnostic.SA1200.severity = none
+dotnet_diagnostic.SA1201.severity = none
+dotnet_diagnostic.SA1202.severity = warning
+dotnet_diagnostic.SA1203.severity = warning
+dotnet_diagnostic.SA1204.severity = warning
+dotnet_diagnostic.SA1205.severity = none
+dotnet_diagnostic.SA1206.severity = warning
+dotnet_diagnostic.SA1207.severity = warning
+dotnet_diagnostic.SA1208.severity = warning
+dotnet_diagnostic.SA1209.severity = warning
+dotnet_diagnostic.SA1210.severity = warning
+dotnet_diagnostic.SA1211.severity = warning
+dotnet_diagnostic.SA1212.severity = warning
+dotnet_diagnostic.SA1213.severity = none
+dotnet_diagnostic.SA1214.severity = warning
+dotnet_diagnostic.SA1215.severity = warning
+dotnet_diagnostic.SA1216.severity = warning
+dotnet_diagnostic.SA1217.severity = warning
+
+# Naming Rules (SA1300-) https://github.com/DotNetAnalyzers/StyleCopAnalyzers/blob/1.1.118/documentation/NamingRules.md
+
+dotnet_diagnostic.SA1300.severity = warning
+dotnet_diagnostic.SA1301.severity = warning
+dotnet_diagnostic.SA1302.severity = warning
+dotnet_diagnostic.SA1303.severity = warning
+dotnet_diagnostic.SA1304.severity = warning
+dotnet_diagnostic.SA1305.severity = warning
+dotnet_diagnostic.SA1306.severity = warning
+dotnet_diagnostic.SA1307.severity = warning
+dotnet_diagnostic.SA1308.severity = warning
+dotnet_diagnostic.SA1309.severity = warning
+dotnet_diagnostic.SA1310.severity = none
+dotnet_diagnostic.SA1311.severity = warning
+dotnet_diagnostic.SA1312.severity = warning
+dotnet_diagnostic.SA1313.severity = warning
+dotnet_diagnostic.SA1314.severity = warning
+
+# Maintainability Rules (SA1400-) https://github.com/DotNetAnalyzers/StyleCopAnalyzers/blob/1.1.118/documentation/MaintainabilityRules.md
+
+dotnet_diagnostic.SA1400.severity = warning
+dotnet_diagnostic.SA1401.severity = warning
+dotnet_diagnostic.SA1402.severity = warning
+dotnet_diagnostic.SA1403.severity = warning
+dotnet_diagnostic.SA1404.severity = warning
+dotnet_diagnostic.SA1405.severity = warning
+dotnet_diagnostic.SA1406.severity = warning
+dotnet_diagnostic.SA1407.severity = warning
+dotnet_diagnostic.SA1408.severity = warning
+# SA1409 is not part of StyleCop.Analyzers
+dotnet_diagnostic.SA1410.severity = warning
+dotnet_diagnostic.SA1411.severity = warning
+dotnet_diagnostic.SA1412.severity = none
```

Comment:
I thinks this one is useful, as BOM is generally useful. It would help us to deal with the "no change" file in Git history marked as changed. We can fix all files in a single PR with some short script.